openedx2zim/scraper.py

```diff
@@ -509,6 +510,33 @@ def dl_dependencies(self, content, output_path, path_from_html):
             content = lxml.html.tostring(html_body, encoding="unicode")
         return content
 
+    def get_favicon(self):
+        """ get the favicon from the given URL for the instance or the fallback URL """
+
+        favicon_fpath = self.build_dir.joinpath("favicon.png")
+        favicon_url = self.instance_connection.instance_config["favicon_url"]
+
+        # fallback method if favicon_url is not specified
+        if favicon_url is None:
```

Comment:
This sounds way too complicated for what it does.
Could you just have something like

```py
for favicon_url in [self.instance_connection.instance_config["favicon_url"], "https://github.com/edx/edx-platform/raw/master/lms/static/images/favicon.ico"]:
    try:
        save_large_file(favicon_url, favicon_fpath)
        break
    except Exception:
        continue
if not favicon_fpath.exists():
    raise Exception("no favicon")
````

Reply:
Okay. Yup. Much cooler way to do that.